simulator/rsa_sighash_all_usesim.c

```diff
@@ -114,36 +113,63 @@ int main(int argc, const char *argv[]) {
     goto exit;
   }
 
-  info.key_size = 2048;
-  unsigned char N2048[2048 / 8];
-  dup_buffer(N_buff, 1024 / 8, N2048, 2);
-  info.N = N2048;
-  int result3 = validate_signature(NULL, (const uint8_t *)&info, sizeof(info),
-                                   (const uint8_t *)msg, strlen(msg), &output,
-                                   &output_len);
-  if (result3 == ERROR_RSA_VERIFY_FAILED) {
-    mbedtls_printf("validate signature (2048-bit) passed\n");
-  } else {
-    mbedtls_printf("validate signature (2048-bit) failed: %d\n", result);
-    exit_code = ERROR_RSA_VERIFY_FAILED;
-    goto exit;
+  {
+    int key_size = 2048;
+    uint32_t length = calculate_rsa_info_length(key_size);
+    uint8_t info_buff[length];
+    RsaInfo *info = (RsaInfo*)info_buff;
```

Comment:
Hmm I think we are still relying on RsaInfo struct layout here? I would suggest we do hand parsing like [this](https://github.com/nervosnetwork/ckb-miscellaneous-scripts/pull/6/files#diff-f4d83f5a4c79b398b0fde13cad97e5b7R180-R185), so the outside code, possibly in other languages, do not need knowledge on C struct layout